libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -93,12 +93,47 @@ proc resetMessage(s: LPChannel) {.async.} =
   await s.conn.writeMsg(s.id, s.resetCode)
 
 proc resetByRemote*(s: LPChannel) {.async.} =
-  await allFutures(s.close(), s.closedByRemote())
+  # Immediately block futher calls
   s.isReset = true
-  await s.cleanUp()
+
+  # start and await async teardown
+  let
+    futs = await allFinished(
+      s.close(),
+      s.closedByRemote(),
+      s.cleanUp()
+    )
+
+  for res in futs:
```

Comment:
this seems pretty repetitive, can we perhaps move this to a template, are they sufficiently similar to be able to generalize?